### PR TITLE
Blocks: Add filename param to Tagged Filesink

### PR DESCRIPTION
**Pull request by devnulling**
PR up for testing and discussion.

Fixes: https://github.com/gnuradio/gnuradio/issues/2907

### CODE REVIEW

File: gr-blocks/include/gnuradio/blocks/tagged_file_sink.h [hunk 1]
```diff
@@ -57,8 +57,14 @@ class BLOCKS_API tagged_file_sink : virtual public sync_block
      * \param itemsize The item size of the input data stream.
      * \param samp_rate The sample rate used to determine the time
      *                  difference between bursts
+     * \param filename Ex: 'FileNameKeyword %%Y-%%m-%%d %%H:%%M:%%S.dat' 
+     *                 result as:
+     *                 'FileNameKeyword 2019-12-05 15:51:56.dat'.
```
Comment (mormj) [inline gr-blocks/include/gnuradio/blocks/tagged_file_sink.h]: Can we make the filenames without spaces?

**Comment by mormj** [comment]
I love the idea here - the behavior of the current tagged stream file sink block - default filenames in the current location is a bit problematic.  When running grc the files get dumped to whatever directory you launched grc in.

With this implementation I have a few issues:
1) it won't run - there is a swig exception when I compile and run
2) filenames end up with spaces which I don't prefer
3) directory is still default (ambiguous)

Perhaps we should add an output directory, or allow that to be part of the pathname

but overall I think this idea will be a big improvement on the tagged stream file sink

File: gr-blocks/include/gnuradio/blocks/tagged_file_sink.h (hunk 1 above)
Comment (dkozel) [reply gr-blocks/include/gnuradio/blocks/tagged_file_sink.h]: That's just an example. The default value is an empty string, it isn't immediately clear what name that will result in. The default not having spaces would be a good thing.

Comment (devnulling) [reply gr-blocks/include/gnuradio/blocks/tagged_file_sink.h]: I'll fixup the example, I didn't like this space either.

**Comment by devnulling** [comment]
This currently doesn't work as expect, I ran out of time the other day debugging it, I'll dig into it more and push up the required changes.

**Comment by mbr0wn** [comment]
This PR needs a rebase. @mormj  also had some good issues. I'll close due to lack of activity. Please reopen when it's back on track. Thanks!

**Pull request closed without merging**